dipy/align/streamlinear.py

```diff
@@ -910,29 +924,39 @@ def check_range(streamline, gt=greater_than, lt=less_than):
 
     if select_random is not None:
         rstreamlines1 = select_random_set_of_streamlines(streamlines1,
-                                                         select_random)
+                                                         select_random,
+                                                         rng=rng)
     else:
         rstreamlines1 = streamlines1
 
     rstreamlines1 = set_number_of_points(rstreamlines1, nb_pts)
-    qb1 = QuickBundles(threshold=qb_thr)
-    rstreamlines1 = [s.astype('f4') for s in rstreamlines1]
-    cluster_map1 = qb1.cluster(rstreamlines1)
+
+    # qb1 = QuickBundles(threshold=qb_thr)
+    rstreamlines1._data.astype('f4')
+    '''#rstreamlines1 = [s.astype('f4') for s in rstreamlines1]
+    # cluster_map1 = qb1.cluster(rstreamlines1)'''
+
+    cluster_map1 = qbx_and_merge(rstreamlines1, thresholds=qbx_thr, rng=rng)
     clusters1 = remove_clusters_by_size(cluster_map1, rm_small_clusters)
-    qb_centroids1 = [cluster.centroid for cluster in clusters1]
+
+    qb_centroids1 = clusters1
 
     if select_random is not None:
         rstreamlines2 = select_random_set_of_streamlines(streamlines2,
-                                                         select_random)
+                                                         select_random,
+                                                         rng=rng)
     else:
         rstreamlines2 = streamlines2
 
     rstreamlines2 = set_number_of_points(rstreamlines2, nb_pts)
-    qb2 = QuickBundles(threshold=qb_thr)
-    rstreamlines2 = [s.astype('f4') for s in rstreamlines2]
-    cluster_map2 = qb2.cluster(rstreamlines2)
+    rstreamlines2._data.astype('f4')
+    '''# qb2 = QuickBundles(threshold=qb_thr)
+    #rstreamlines2 = [s.astype('f4') for s in rstreamlines2]
+    # cluster_map2 = qb2.cluster(rstreamlines2)'''
+    cluster_map2 = qbx_and_merge(rstreamlines2, thresholds=qbx_thr, rng=rng)
+
     clusters2 = remove_clusters_by_size(cluster_map2, rm_small_clusters)
-    qb_centroids2 = [cluster.centroid for cluster in clusters2]
+    qb_centroids2 = clusters2
```

Comment:
Why you do not do directly `qb_centroids2 = remove_clusters_by_size(cluster_map2, rm_small_clusters)`?

`clusters2` variable seems useless